src/components/calculator/PointsDisplay.tsx

```diff
@@ -78,77 +82,150 @@ export function ExplanationCard(props: {
-        )}% of your annual risk
-        allocation.`
+const RISK_LEVELS_TO_SHOW_ON_LEGEND = 5 // Shows up through 'Very High'
+
+function howRisky(points: number, budget: number): RiskLevel {
+  // First check against dangerous risk levels. Don't normalize points here because we primarily want to indicate the risk to others, not the risk to you others at these "dangerous" levels
+  const highestNormalRisklevel = riskLevels[riskLevels.length - 1]
+  if (points >= highestNormalRisklevel.max) {
+    return lifeThreateningRisk
   }
-  const weekBudget = budget / 50 // Numbers look cleaner than 52.
-  if (points > weekBudget) {
-    const weeksConsumed = fixedPointPrecision(points / weekBudget)
-    return `Doing this activity once would use up your entire risk allocation for
-        ~${weeksConsumed} ${
-      Number.parseInt(weeksConsumed) > 1 ? 'weeks' : 'week'
-    }.`
+
+  // Then check against normalized points
+  const normalizedPoints = points / (budget / 10000)
+  const curLevel = riskLevels.find((level) => normalizedPoints < level.max)
+  return (
+    curLevel || riskLevels[riskLevels.length - 1] // Default to the highest risk level
+  )
+}
+
+const getWeekBudget = (budget: number) => {
+  return budget / 50 // Numbers look cleaner than 52.
+}
+
+const budgetConsumption = (points: number, budget: number) => {
+  const weekBudget = getWeekBudget(budget)
+  const weeksConsumed = points / weekBudget
+  if (weeksConsumed >= 1.5) {
+    return `
+        ${fixedPointPrecision(weeksConsumed)}x  your weekly risk budget`
   }
-  return `Doing this activity once would use up
-      ~${fixedPointPrecision((points / weekBudget) * 100)}% of your risk
-      allocation for one week.`
+  return `${fixedPointPrecision(
+    (points / weekBudget) * 100,
+  )}% of your weekly risk budget`
 }
 
 export function PointsDisplay(props: {
   points: number
   repeatedEvent: boolean
+  riskBudget: number
+  riskBudgetSetter: (newValue: number) => void
   upperBound: number
   lowerBound: number
 }): React.ReactElement {
+  const currentRiskLevel = howRisky(props.points, props.riskBudget)
+  const doShowPoints = showPoints(props.points)
   return (
-    <div className="top-half-card">
-      <strong>Results:</strong>
-      {showPoints(props.points) ? (
-        <h1>
-          {displayPoints(props.points)} microCOVIDs (
-          {displayPoints(props.lowerBound)} to {displayPoints(props.upperBound)}
-          ){props.repeatedEvent ? ' per week' : ' each time'}
-        </h1>
-      ) : (
-        <h1>fill in calculator to see</h1>
-      )}
-    </div>
+    <Row className="top-half-card no-gutters">
+      <Col className="legend-container">
+        {doShowPoints && currentRiskLevel.icon ? (
+          <>
+            <currentRiskLevel.icon
+              className={'risk-icon text-risk-' + currentRiskLevel.style}
+            />
+          </>
+        ) : (
+          riskLevels
+            .map((level) => level.style)
```

Comment:
Moved it out into a series of helper functions and its own component to make it more clear. Also renamed "legend" to "thermometer" because it is more obvious